examples/java_postgresql_mixed_binding_type/README.md

```diff
@@ -0,0 +1,269 @@
+
+* `Git Repo URL` = `https://github.com/ldimaggi/java-rest-http-crud`
+* `Project` = `service-binding-demo`
+* `Application`->`Create New Application` = `java-app`
+* `Name` = `java-app`
+* `Builder Image` = `Java`
+* `Create a route to the application` = checked
+* `Select the resource type to generate` = Deployment
+
+and click on the `[Create]` button.
+
+Notice, that during the import no DB config was mentioned or requestd.
+
+When the application is running navigate to its route to verify that it is up. Try the application's UI to add a fruit - it causes an error proving that the DB is not connected.
+
+#### Create a DB instance for the application
+
+Now we utilize the DB operator that the cluster admin has installed. To create a DB instance just create a `Database` custom resource in the `service-binding-demo` namespace called `db-demo`:
+
+```shell
+cat <<EOS |kubectl apply -f -
+---
+apiVersion: postgresql.baiju.dev/v1alpha1
+kind: Database
+metadata:
+  name: db-demo
+  namespace: service-binding-demo
+spec:
+  image: docker.io/postgres
+  imageName: postgres
+  dbName: db-demo
+  tags:
+	- "centos7-12.3"
+        - "centos7-12.4"
+  storage:
+    - primaryStorage:
+        - "storageClass"
+        - 1000
+  userLabels:
+    archive: "false"
+    environment: "demo"   
+  secretName:
+     - primarySecretName: "example-primaryuser"
+     - rootSecretName: "example-secretuser"
+EOS
+```
+
+Alternatively, you can perform the same task with this make command:
+
+```shell
+make create-backing-db-instance
+```
+
+#### Express an intent to bind the DB and the application
+
+Now, the only thing that remains is to connect the DB and the application. We let the Service Binding Operator to 'magically' do the connection for us.
+
+Create the following `ServiceBindingRequest`:
+
+```shell
+cat <<EOS |kubectl apply -f -
+---
+apiVersion: apps.openshift.io/v1alpha1
+kind: ServiceBindingRequest
+metadata:
+  name: binding-request
+  namespace: service-binding-demo
+spec:
+  applicationSelector:
+    resourceRef: java-app
+    group: apps
+    version: v1
+    resource: deployments
+  backingServiceSelector:
+    group: postgresql.baiju.dev
+    version: v1alpha1
+    kind: Database
+    resourceRef: db-demo
+    id: postgresDB
+ customEnvVar:
+      - name: JDBC_URL
+        value: 'jdbc:postgresql://{{ .postgresDB.status.dbConnectionIP }}:{{ .postgresDB.status.dbConnectionPort }}/{{ .postgresDB.status.dbName }}'
+      - name: DB_USER
+        value: '{{ .postgresDB.status.dbCredentials.user }}'
+      - name: DB_PASSWORD
+        value: '{{ .postgresDB.status.dbCredentials.password }}'
+      - name: TAGS
+        value: '{{ .postgresDB.spec.tags.value1.centos7-12.3 }}'
+      - name: STORAGE
+        value: '{{ .postgresDB.spec.storage }}'
+      - name: USERLABELS
+        value: '{{ .postgresDB.spec.userlabels }}'
+      - name: SECRETNAME
+        value: '{{ .postgresDB.spec.secretName }}'
```

Comment:
What are you planning to store in `SECRETNAME`? The field holds a `[]map[string]interface{}`, and will produce by default `[map[primarySecretName:example-primaryuser] map[rootSecretName:example-secretuser]]` which might not be what you intend.

See https://play.golang.org/p/XtAWhG8EbCX.

Reply:
I get the environment variables as:

```
DATABASE_SECRETNAME_0_PRIMARYSECRETNAME=example-primaryuser
DATABASE_SECRETNAME_0_SECONDARYSECRETNAME=example-secondaryuser
DATABASE_SECRETNAME_1_ROOTSECRETNAME=example-secretuser
```

which is what I intend to get.